mixins/test/label-mixin.test.js

```diff
@@ -0,0 +1,162 @@
+
+import { defineCE, expect, fixture, html, nextFrame } from '@open-wc/testing';
+import { LabelledMixin, LabelMixin } from '../labelled-mixin.js';
+import { LitElement } from 'lit-element/lit-element.js';
+
+const labelledTag = defineCE(
+	class extends LabelledMixin(LitElement) {
+		render() {
+			return html`
+				<input type="text" aria-label="${this.label}">
+			`;
+		}
+	}
+);
+
+const labelTag = defineCE(
+	class extends LabelMixin(LitElement) {
+		static get properties() {
+			return {
+				text: { type: String }
+			};
+		}
+		render() {
+			return html`
+				<span>${this.text}</span>
+			`;
+		}
+		updated(changedProperties) {
+			super.updated(changedProperties);
+			this.updateLabel(this.text);
+		}
+	}
+);
+
+describe.only('LabelMixin', () => {
+
+	let elem;
+
+	beforeEach(async() => {
+		elem = await fixture(`
+			<${labelTag} text="the label value"></${labelTag}>
+		`);
+	});
+
+	it('reflects label value', async() => {
+		expect(elem.getAttribute('_label')).to.equal('the label value');
+	});
+
+	it('reflects label value change', async() => {
+		elem.text = 'new label value';
+		await nextFrame();
+		expect(elem.getAttribute('_label')).to.equal('new label value');
+	});
+
+});
+
+describe.only('LabelledMixin', () => {
+
+	let elem;
+
+	beforeEach(async() => {
+		elem = await fixture(`
+			<div>
+				<table>
+					<tr>
+						<td><${labelledTag} labelled-by="label1"></${labelledTag}></td>
+						<td><span id="label1">native element</span></td>
+					</tr>
+					<tr>
+						<td><${labelledTag} label="explicit label"></${labelledTag}></td>
+						<td></td>
+					</tr>
+					<tr>
+						<td><${labelledTag} labelled-by="label2"></${labelledTag}></td>
+						<td><${labelTag} id="label2" text="custom elemnt"></${labelTag}></td>
+					</tr>
+				</table>
+				<span id="label3">other element</span>
+			</div>
+		`);
+	});
+
```

Comment:
For all three of these cases, it might be good to test some of the negative cases -- like `id`s referencing elements that don't exist, or elements that are not in the same DOM scope. Just to make sure that your error handling/logging code isn't blowing up.